mathjax3-ts/input/tex/ParseUtil.ts

```diff
@@ -488,6 +488,110 @@ namespace ParseUtil {
     return (font ? {mathvariant: font} : {});
   };
 
+
+  /**
+   * Splits a package option list of the form [x=y,z=1] into an attribute list
+   * of the form {x: y, z: 1}.
+   * @param {string} attrib The attributes of the package.
+   * @param {{[key: string]: number}?} allowed A list of allowed options. If
+   *     given only allowed arguments are returned.
+   * @param {boolean?} error If true, raises an exception if not allowed options
+   *     are found.
+   * @return {EnvList} The attribute list.
+   */
+  export function keyvalOptions(attrib: string,
+                                allowed: {[key: string]: number} = null,
+                                error: boolean = false): EnvList {
+    let def: EnvList = readKeyval(attrib);
+    if (allowed) {
+      for (let key of Object.keys(def)) {
+        if (!allowed[key]) {
```

Comment:
This still needs to be 

```suggestion
        if (!allowed.hasOwnProperty(key)) {
```

since otherwise it always allows `toString`, `toSource`, valueOf` or any of the other standard properties of an `Object` to be specified, which we don't want.